python/dolfinx/fem/function.py

```diff
@@ -271,6 +271,14 @@ def __init__(self, V: FunctionSpace, x: typing.Optional[la.Vector] = None,
             if dtype is None:
                 dtype = default_scalar_type
 
+        element_type = np.dtype(type(V.element).__name__.split("_")[1])
```

Comment:
would be good to add a `dtype` property to the classes so that we can get this info without needing to check the type name string

Reply:
> would be good to add a `dtype` property to the classes so that we can get this info without needing to check the type name string

Yes, we can do this. This simplest way it to return a `char` from the nanobind layer that indicates the type, see https://numpy.org/doc/stable/reference/arrays.scalars.html#floating-point-types.